libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts

```diff
@@ -1050,4 +1067,86 @@ export const buildTigerSpecificFunctions = (
             });
         });
     },
+    getAllCSPDirectives: async (): Promise<Array<ICSPDirective>> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getAllEntitiesCspDirectives({});
+                return result.data?.data || [];
+            });
+        } catch (error) {
+            return [];
+        }
+    },
+    getCSPDirective: async (directiveId: string): Promise<ICSPDirective> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getEntityCspDirectives({ id: directiveId });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            throw convertApiError(error);
+        }
+    },
+    createCSPDirective: async (requestData: ICSPDirective): Promise<ICSPDirective> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const jsonApiCspDirectiveInDocument: JsonApiCspDirectiveInDocument = {
+                    data: {
+                        id: requestData.id,
+                        type: JsonApiCspDirectiveInTypeEnum.CSP_DIRECTIVE,
+                        attributes: requestData.attributes,
+                    },
+                };
+                const result = await sdk.entities.createEntityCspDirectives({
+                    jsonApiCspDirectiveInDocument,
+                });
+
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            if (error.response?.status === 400) {
+                const message = error?.response?.data?.detail
+                    ? error?.response?.data?.detail
+                    : "Server error";
+                throw new UnexpectedError(message, error);
+            }
+            throw convertApiError(error);
+        }
+    },
+    updateCSPDirective: async (directiveId: string, requestData: ICSPDirective): Promise<ICSPDirective> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const jsonApiCspDirectiveInDocument: JsonApiCspDirectiveInDocument = {
+                    data: {
+                        id: requestData.id,
+                        type: JsonApiCspDirectiveInTypeEnum.CSP_DIRECTIVE,
+                        attributes: requestData.attributes,
+                    },
+                };
+                const result = await sdk.entities.updateEntityCspDirectives({
+                    id: directiveId,
+                    jsonApiCspDirectiveInDocument,
+                });
+
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            if (error.response?.status === 400) {
+                const message = error?.response?.data?.detail
+                    ? error?.response?.data?.detail
+                    : "Server error";
```

Comment:
Status code 400 means Bad request, doesn't it?

→ Using `"Server error"` here seems to be unsuitable. Please change it to `"An unexpected error has occurred"` or `"Bad request"` instead.